tests/test_base_scalers.py

```diff
@@ -0,0 +1,114 @@
+from unittest.mock import patch
+import json
+import os
+
+import pytest
+
+from app.base_scalers import BaseScaler, AwsBaseScaler, DbQueryScaler
+
+
+class TestBaseScaler:
+    def test_init_assigns_basic_values(self):
+        input_attrs = {
+            'min_instances': 1,
+            'max_instances': 2,
+            'threshold': 1500,
+            'extra_field_1': 'some_value'
+        }
+        base_scaler = BaseScaler(**input_attrs)
+
+        assert base_scaler.min_instances == input_attrs['min_instances']
+        assert base_scaler.max_instances == input_attrs['max_instances']
+        assert base_scaler.threshold == input_attrs['threshold']
+
+    @pytest.mark.parametrize('min_instances,max_instances,desired_instances,expected_instances', [
+        (3, 5, 4, 4),
+        (3, 5, 7, 5),
+        (3, 5, 2, 3),
+        (3, 5, -1, 3),
```

Comment:
quick comment or `ids=[]` explaining each test version would be nice eg "scale up", "scale up but above max" etc etc